api/openapi_server/test/__init__.py

```diff
@@ -2,81 +2,67 @@
 from pathlib import Path
 
 import connexion
-from flask import json
 from flask_testing import TestCase
 from typing import List
-from werkzeug.test import TestResponse
 
 from openapi_server.encoder import JSONEncoder
-from openapi_server.models import database
+from openapi_server.models.database import DataAccessLayer
 from openapi_server.__main__ import get_bundled_specs
+from openapi_server.repositories.service_provider_repository import HousingProviderRepository
 
 
 class BaseTestCase(TestCase):
 
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.cached_engine = None
-        self.tmp_testing_database = Path("testinghomeuniteus.db")
-        
-
     def create_app(self):
         '''
-        Create a temporary, empty database for testing purposes and setup
-        a new instance of our Flask App for testing purposes.
-        '''
-        # Cache a reference to the current engine, so we can restore the previous
-        # DataAccessLayer once our test is complete
-        self.cached_engine = database.DataAccessLayer._engine
+        Create a temporary, empty database for testing purposes and return
+        a new instance of our Flask App to the base class for testing purposes.
+
+        The base class will never start the Flask App. It instead create a
+        mock self.client class that is used to simulate requests to the WSGI server.
 
-        database.DataAccessLayer._engine = None 
-        database.DataAccessLayer.get_engine(f"sqlite:///./{self.tmp_testing_database}")
-        database.DataAccessLayer.db_init()
+        https://flask.palletsprojects.com/en/2.2.x/testing/
+        https://werkzeug.palletsprojects.com/en/2.3.x/test/
+        '''
+        # Create a temporary, memory only database. This temp db will be
+        # automatically destroyed when the refcount drops to zero
+        DataAccessLayer._engine = None
+        DataAccessLayer._conn_string = "sqlite:///:memory:"
+        DataAccessLayer.db_init()
+        self.provider_repo = HousingProviderRepository(DataAccessLayer.get_engine())
 
         logging.getLogger('connexion.operation').setLevel('ERROR')
         app = connexion.App(__name__)
         app.app.json_encoder = JSONEncoder
         app.add_api(get_bundled_specs(Path('openapi_server/openapi/openapi.yaml')),
                 arguments={'title': 'Home Unite Us'},
```

Comment:
Unrelated to this PR: Turns out, this isn't used. The `arguments` argument is used to populate template variables in the openapi definition. However, there's no such template variable `title` in the openapi.yaml.

Reply:
Great catch. I removed it in [1d8a319](https://github.com/hackforla/HomeUniteUs/pull/575/commits/1d8a319d4d549ac1424390e5c468d8689aba3f97) since it doesn't do anything